### PR TITLE
account.listaccounts, account.getaccount, account.gettokenbalances, account.listaccounthistory

### PR DESCRIPTION
**Pull request by canonbrother**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Implemented type for RPC: #48
1. account.listaccounts
2. account.getaccount
3. account.gettokenbalances
4. account.listaccounthistory

### CODE REVIEW

File: packages/jellyfish-api-core/src/category/account.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
+import BigNumber from 'bignumber.js'
+import { ApiClient } from '../.'
+
+type OwnerType = 'mine' | 'all' | string
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/src/category/account.ts:4]:
```
/**
  * Single account ID (CScript or address) or reserved words, 
  * 'mine' to list history for all owned accounts or 
  * 'all' to list the whole DB
  */
```

Add docs for this.

**Review by fuxingloh — CHANGES_REQUESTED** [review]
great work, I wrote some minor changes to consider

File: packages/jellyfish-api-core/__tests__/category/account.test.ts [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+  }
+
+  describe('listAccounts', () => {
+    it('should listAccounts', async () => {
+      await waitForExpect(async () => {
+        const accounts = await client.account.listAccounts({}, true, { indexedAmounts: false, isMineOnly: false })
+        expect(accounts.length).toBeGreaterThan(0)
+      })
+
+      const accounts = await client.account.listAccounts({}, true, { indexedAmounts: false, isMineOnly: false })
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/__tests__/category/account.test.ts:78]: and for this type of test, you should test without setting the default options. or else you miss test coverage for this.

https://app.codecov.io/gh/DeFiCh/jellyfish/compare/162/diff#diff-cGFja2FnZXMvamVsbHlmaXNoLWFwaS1jb3JlL3NyYy9jYXRlZ29yeS9hY2NvdW50LnRz

This applyies for all the default cases.

File: packages/jellyfish-api-core/__tests__/category/account.test.ts [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+    await accountToAccount('DETH', 46, from)
+  }
+
+  async function utxosToAccount (amount: number): Promise<void> {
+    const address = await container.call('getnewaddress')
+
+    const payload: { [key: string]: string } = {}
+    payload[address] = `${amount.toString()}@0`
+    await container.generate(25)
+  }
```
[the remark this answers is not in the document]
Comment (canonbrother) [reply packages/jellyfish-api-core/__tests__/category/account.test.ts:44]: hmm.. seems i understand the relationship between account and address wrongly.
yea its not needed

File: packages/jellyfish-api-core/__tests__/category/account.test.ts (hunk 2 above)
Comment (canonbrother) [reply packages/jellyfish-api-core/__tests__/category/account.test.ts]: phew.. i guess setting the method with optional paras on first order is a rule, its more flexible

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**